Docker-dev-build

```diff
@@ -0,0 +1,135 @@
+      spack test run --alias test_serial octopus && \
+      # display output from smoke tests (just for information):
+      spack test results -l test_serial && \
+      # show which octopus version we use (for convenience):
+      spack load octopus && octopus --version && \
+      # Run make checks
+      cd ~/dev-build-serial ; \
+      echo "-+-+-+-+ Running make check-short" ; \
+      make check-short | tee serial_check-short.log || /bin/true ; \
+      echo "-+-+-+-+ Running make check-long" ; \
+      make check-long  | tee serial_check-long.log || /bin/true ; \
+      echo "-+-+-+-+ Parsing serial_check-short.log" ; \
+      python3 /home/user/check_buildlog.py serial_check-short.log foss2021a-serial || /bin/true; \
+      echo "-+-+-+-+ Parsing serial_check-long.log" ; \
+      python3 /home/user/check_buildlog.py serial_check-long.log foss2021a-serial || /bin/true; \
+      # deactivate the environment.
+      spack env deactivate
+
+# # MPI version
+
+RUN . $SPACK_ROOT/share/spack/setup-env.sh && \
+      # create a new environment for the MPI version and activate it:
+      spack env create octopus-mpi && \
+      spack env activate octopus-mpi && \
+      # make a dev-build direcotry for the parallel version:
+      mkdir ~/dev-build-parallel && cd ~/dev-build-parallel && \
+      wget https://octopus-code.org/down.php?file=12.1/octopus-12.1.tar.gz -O octopus-12.1.tar.gz &&\
+      tar --strip-components=1 -xf octopus-12.1.tar.gz -C ./&& \
+      # display specs of upcoming spack installation:
+      spack spec octopus@${OCT_VERSION} +mpi +netcdf+parmetis+arpack+cgal+pfft+python+likwid+libyaml+elpa+nlopt~debug~cuda~metis~scalapack  && \
+      # run the spack installation (adding it to the environment):
+      spack dev-build octopus@${OCT_VERSION} +mpi +netcdf+parmetis+arpack+cgal+pfft+python+likwid+libyaml+elpa+nlopt~debug~cuda~metis~scalapack  && \
+      # run spack smoke tests for octopus. We get an error if any of the fails:
+      spack test run --alias test_MPI octopus && \
+      # display output from smoke tests (just for information):
+      spack test results -l test_MPI && \
+      # show which octopus version we use (for convenience):
+      spack load octopus && octopus --version && \
+      # Run make checks
+      cd ~/dev-build-parallel ; \
+      echo "+|+|+|+ Running make check-short" ; \
+      make check-short | tee mpi_check-short.log || /bin/true ; \
+      echo "+|+|+|+ Running make check-long" ; \
+      make check-long  | tee mpi_check-long.log || /bin/true ; \
+      echo "+|+|+|+ Parsing mpi_check-short.log" ; \
+      python3 /home/user/check_buildlog.py mpi_check-short.log foss2021a-mpi || /bin/true; \
+      echo "+|+|+|+ Parsing mpi_check-long.log" ; \
+      python3 /home/user/check_buildlog.py mpi_check-long.log foss2021a-mpi || /bin/true; \
+      spack env deactivate
+
+# Provide bash in case the image is meant to be used interactively
```

Comment:
Would be nice to have a summary of passed and failed tests (if easily done) for serial and mpi. So that you can scroll to the end and find the most interesting information there.

Or perhaps just re-run the `check_buildog.py` script at the very end.

Reply:
Currently `check_buildog.py` is the last thing thats run for each variant.
It gives the summary in the following format:

```bash
-+-+-+-+ Parsing serial_check-short.log
Unexpected failures :
periodic_systems/24-hartree_fock_1D.test, functionals/18-mgga.test, finite_systems_3d/37-sternheimer_polarized.test, finite_systems_3d/06-restart.test, finite_systems_1d/01-hydrogen.test
Built with tool chain : foss2021a-serial
-+-+-+-+ Parsing serial_check-long.log
Unexpected failures :
real_time/14-absorption-spinors.test, periodic_systems/21-magnon.test, periodic_systems/07-mgga.test, linear_response/01-casida.test, finite_systems_3d/16-scfinlcao_std.test
Built with tool chain : foss2021a-serial

------------------ Build log and check log for MPI variant ---------------


+|+|+|+ Parsing mpi_check-short.log
Unexpected failures :
real_time/12-absorption.test, periodic_systems/24-hartree_fock_1D.test, functionals/18-mgga.test, finite_systems_3d/37-sternheimer_polarized.test, finite_systems_3d/06-restart.test
Built with tool chain : foss2021a-mpi
+|+|+|+ Parsing mpi_check-long.log
Unexpected failures :
real_time/14-absorption-spinors.test, periodic_systems/07-mgga.test, linear_response/01-casida.test, finite_systems_3d/16-scfinlcao_std.test
Built with tool chain : foss2021a-mpi
```